### ISSUE TITLE
The performance of SparseSoftmaxCrossEntropyWithLogits in inference is very low

### ISSUE THREAD

**Issue by mpjlu** [tensorflow/tensorflow#25958]
SparseSoftmaxCrossEntropyWithLogits computes softmax cross entropy cost and gradients to backpropagate. 
When SparseSoftmaxCrossEntropyWithLogits used in inference, for example, in language models, we only need the cost. 
SparseSoftmaxCrossEntropyWithLogits is very time-consuming in language models, it is better to add a flag for inference and training, or to split this OP to two.

**Describe the current behavior**
Profile data of language model:
node name | requested bytes | total execution time | accelerator execution time | cpu execution time
**SparseSoftmaxCrossEntropyWithLogits     245.76KB (100.00%, 0.00%),      1.58sec (100.00%, 51.51%),             0us (0.00%, 0.00%),      1.58sec (100.00%, 51.51%)**
MatMul                      8905.28MB (100.00%, 88.05%),       1.10sec (48.49%, 36.00%),             0us (0.00%, 0.00%),       1.10sec (48.49%, 36.00%)
Add                                 84B (11.95%, 0.00%),       201.16ms (12.49%, 6.57%),             0us (0.00%, 0.00%),       201.16ms (12.49%, 6.57%)
Split                          671.09MB (11.95%, 6.64%),        105.97ms (5.92%, 3.46%),             0us (0.00%, 0.00%),        105.97ms (5.92%, 3.46%)

**Describe the expected behavior**
Profile data after optimization:
node name | requested bytes | total execution time | accelerator execution time | cpu execution time
MatMul                      8905.28MB (100.00%, 88.05%),      1.11sec (100.00%, 46.86%),             0us (0.00%, 0.00%),      1.11sec (100.00%, 46.86%)
**SparseSoftmaxCrossEntropyWithLogits      245.76KB (11.95%, 0.00%),      871.49ms (53.14%, 36.93%),             0us (0.00%, 0.00%),      871.49ms (53.14%, 36.93%)**
Add                                 84B (11.95%, 0.00%),       200.13ms (16.21%, 8.48%),             0us (0.00%, 0.00%),       200.13ms (16.21%, 8.48%)
Split                          671.09MB (11.95%, 6.64%),        106.76ms (7.73%, 4.52%),             0us (0.00%, 0.00%),        106.76ms (7.73%, 4.52%)

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): el7
- TensorFlow installed from (source or binary):source
- TensorFlow version (use command below):1.12
- Python version:3.7.2
- Bazel version (if compiling from source):0.19.2
- GCC/Compiler version (if compiling from source):gcc6.3

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
https://github.com/mlperf/inference/tree/master/cloud/language_modeling
./benchmark.py

**Comment by mpjlu** [comment]
ping @greg1232

**Comment by skye** [comment]
@azaks2 do you know who's a good person to look at this?

**Comment by mpjlu** [comment]
Hi @skye, @azaks2. I can work on this. Do you think we should optimize this OP?

**Comment by rmothukuru** [comment]
@mpjlu,
Sorry for the delayed response. With so many updates in the **`Tensorflow Code`** and with the release of **`Subsequent Versions`**, I'm pretty confident that this performance issue should be resolved. If possible for you, can you please check it with the **`Latest Tensorflow Version (2.5)`**?
If that is not possible, please provide the reproducible code (the link you provided is not working), so that we can check it.
Thanks!
